erpnext/hr/doctype/salary_slip/salary_slip.py

```diff
@@ -321,9 +321,13 @@ def check_existing(self):
 	def sum_components(self, component_type, total_field):
 		joining_date, relieving_date = frappe.db.get_value("Employee", self.employee,
 			["date_of_joining", "relieving_date"])
+		
 		if not relieving_date:
 			relieving_date = getdate(self.end_date)
 
+		if not joining_date:
```

Comment:
Why this validation is here?
Make joining_date field mandatory instead.

Here only validate if joining_date exists.